src/puppetlabs/puppetdb/middleware.clj

```diff
@@ -265,7 +265,8 @@
   [app reject-large-commands? max-command-size]
   (fn [req]
     (if (= :post (:request-method req))
-      (let [length-in-bytes (request/content-length req)]
+      (let [length-in-bytes (or (request/content-length req)
+                                (Long. (get-in req [:headers "x-uncompressed-length"])))]
```

Comment:
This `Long` constructor can throw an error `java.lang.NumberFormatException`. We should probably guard against that, log a warning message about that problem and use a sane default for `length-in-bytes` (0 is fine by me if that's what it was ending up as before).

And so we are extra confident on this piece of potentially user-input, it might be a good idea to additionally verify that `length-in-bytes` is actually an integer and someone isn't trying to send us π bytes.

Reply:
that's a good point, I'll update it. I think it's ok to let `length-in-bytes` be nil here that's what it has been. If we set it to 0 it will throw off the metrics